cmake/modules/extensions.cmake

```diff
@@ -4845,14 +4849,19 @@ function(add_llext_target target_name)
     DEPENDS ${output_file}
   )
 
-  # Define arch-specific flags for the supported architectures
-  if(CONFIG_ARM)
-    list(PREPEND LLEXT_C_FLAGS "-mlong-calls" "-mthumb")
-  elseif(CONFIG_XTENSA)
-    list(PREPEND LLEXT_C_FLAGS "-shared" "-fPIC" "-nostdlib" "-nodefaultlibs")
-  else()
-    message(FATAL_ERROR "add_llext_target: unsupported architecture")
-  endif()
+  # Convert the LLEXT_REMOVE_FLAGS list to a regular expression, and use it to
+  # filter out these flags from the Zephyr target settings
+  list(TRANSFORM LLEXT_REMOVE_FLAGS
+       REPLACE "(.+)" "^\\1$"
+       OUTPUT_VARIABLE llext_remove_flags_regexp
+  )
+  string(REPLACE ";" "|" llext_remove_flags_regexp "${llext_remove_flags_regexp}")
+  set(zephyr_flags
+      "$<TARGET_PROPERTY:zephyr_interface,INTERFACE_COMPILE_OPTIONS>"
+  )
+  set(zephyr_filtered_flags
+      "$<FILTER:${zephyr_flags},EXCLUDE,${llext_remove_flags_regexp}>"
```

Comment:
This is scary stuff :-) How about adding a couple `message(DEBUG $llext_remove_flags_regexp)` and `message(DEBUG zephyr_filtered_flags)`?
I'm surprised you didn't need them for yourself!

Reply:
I used plenty of them for debugging indeed :innocent: but I did not push them, as they are quite cumbersome: to see the end result to that gibberish you need to add _a `custom_target` that echoes the flags on the console_. The generator syntax is next-level painful to me!

Reply:
Ah yes of course, it's build-time. Got it.

Reply:
cmake is next level painful, thanks for taking this on